conformance/report.go

```diff
@@ -99,3 +155,44 @@ var _ = ReportAfterSuite("MCS conformance report", func(report Report) {
 	err = tmpl.Execute(out, data)
 	Expect(err).To(Succeed())
 })
+
+func parseFailureMessage(s string) string {
+	// First see if the message represents an error formatted by a gomega matcher - we're interested in
+	// extracting the optional user description passed to the gomega assertion and the actual error
+	// string as these are the useful parts conducive for formatting in the report table.
+	matches := errorRegEx.FindStringSubmatch(s)
+	if len(matches) > 0 {
+		// First match at index 0 is the full text that was matched; index 1 will be the user description
+		// and index 2 the error string. We concatenate the latter two.
+		msg := strings.TrimSpace(matches[1])
+		if msg == "" {
+			msg = strings.TrimSpace(matches[2])
+		} else {
+			msg = msg + ": " + strings.TrimSpace(matches[2])
+		}
+
+		return msg
+	}
+
+	// Fallback - just take the first line in the message.
+	return firstLine(s)
+}
+
+func firstLine(s string) string {
+	sc := bufio.NewScanner(strings.NewReader(strings.TrimSpace(s)))
+	if sc.Scan() {
+		return strings.TrimSpace(sc.Text())
+	}
+
+	return ""
```

Comment:
Since the strings parsed here are built from string constants (from Ginkgo and elsewhere), this should be safe:

```suggestion
	first, _, _ := strings.Cut(s, "\n")
	return strings.TrimSpace(first)
```

(See https://golang.org/ref/spec#String_literals for context.)

Reply:
OK - I was trying to be utmost safe.